dts/bindings/gpio/gpio-generic.yaml

```diff
@@ -0,0 +1,21 @@
+title: GPIO GENERIC
+
+description: >
+    This is a representation of the GPIO GENERIC nodes
+
+properties:
+    compatible:
+      constraint: "gpio-generic"
+      type: string-array
+      category: required
+      description: compatible strings
+
+sub-node:
+    properties:
+       gpios:
+          type: compound
+          category: required
+       label:
+          category: required
+          type: string
+          description: Human readable string describing the device (used by Zephyr for API name)
```

Comment:
Wonder if this description being copy-pasted in lots of bindings is overkill.

Maybe we could start cleaning up inconsistent indent, like this:

```
sub-node:
    properties:
        gpios:
            type: compound
            category: required

        label:
            type: string
            category: required
```

Reply:
So 4 spaces is what we want for indentation?

Reply:
I like 4 spaces. Nice to have it consistent at least, instead of 4 spaces for the first level and then 2.

Reply:
> Can you give an example of when or how this would be used. Normally I'd expect GPIOs to be related to some other device.

A lot of peripheral devices have very simple interfaces. For example a motion sensor as an input or a relay control pin as output. I think it makes more sense to have these as generic gpios rather than having a separate binding.